src/containers/SearchPage/components/form/SearchForm.tsx

```diff
@@ -55,30 +67,36 @@ export const parseSearchParams = (locationSearch: string): SearchParams => {
-    language: queryStringObject.language,
-    license: queryStringObject.license,
-    page: parseNumberParam("page"),
-    "page-size": parseNumberParam("page-size"),
-    sort: queryStringObject.sort,
-    status: queryStringObject.status,
-    subjects: queryStringObject.subjects,
-    type: queryStringObject.type,
-    users: queryStringObject.users,
-    "revision-date-from": queryStringObject["revision-date-from"],
-    "revision-date-to": queryStringObject["revision-date-to"],
-    "exclude-revision-log": parseBooleanParam("exclude-revision-log"),
-    "responsible-ids": queryStringObject["responsible-ids"],
-    "filter-inactive": parseBooleanParam("filter-inactive", true),
+  const searchBodyKeyMapping: Record<keyof SearchParams, { key: keyof SearchParamsBody; data: unknown }> = {
+    "draft-status": { key: "draftStatus", data: queryStringObject["draft-status"]?.split(",") },
+    "include-other-statuses": { key: "includeOtherStatuses", data: parseBooleanParam("include-other-statuses") },
+    "resource-types": { key: "resourceTypes", data: queryStringObject["resource-types"]?.split(",") },
+    "audio-type": { key: "audioType", data: queryStringObject["audio-type"] },
+    "concept-type": { key: "conceptType", data: queryStringObject["concept-type"] },
+    "model-released": { key: "modelReleased", data: queryStringObject["model-released"]?.split(",") },
+    "page-size": { key: "pageSize", data: parseNumberParam("page-size") },
+    "revision-date-from": { key: "revisionDateFrom", data: queryStringObject["revision-date-from"] },
+    "revision-date-to": { key: "revisionDateTo", data: queryStringObject["revision-date-to"] },
+    "exclude-revision-log": { key: "excludeRevisionLog", data: parseBooleanParam("exclude-revision-log") },
+    "responsible-ids": { key: "responsibleIds", data: queryStringObject["responsible-ids"]?.split(",") },
+    "filter-inactive": { key: "filterInactive", data: parseBooleanParam("filter-inactive", true) },
+    query: { key: "query", data: queryStringObject.query },
+    language: { key: "language", data: queryStringObject.language },
+    page: { key: "page", data: parseNumberParam("page") },
+    "article-types": { key: "articleTypes", data: queryStringObject["article-types"] },
+    fallback: { key: "fallback", data: parseBooleanParam("fallback") },
+    status: { key: "status", data: queryStringObject.status?.split(",") },
+    sort: { key: "sort", data: queryStringObject.sort },
+    subjects: { key: "subjects", data: queryStringObject.subjects?.split(",") },
+    users: { key: "users", data: queryStringObject.users?.split(",") },
+    license: { key: "license", data: queryStringObject.license },
   };
+  return Object.entries(searchBodyKeyMapping).reduce((result, [key, val]) => {
+    if (!val.data) return result;
+    const updatedKey = parseAsSearchBody ? val.key : key;
+    const updatedVal = val.data;
+    return { ...result, [updatedKey]: updatedVal };
```

Comment:
Her kloner man hele objektet hver eneste iterasjon. Kanskje vi kunne assignet en og en nøkkel? Det er mer effektivt

Reply:
Det gjør typingen her litt mer tricky siden updatedKey kan være av to ulike typer, jeg fant ikke en løsning på hvordan vi burde type det isåfall 😅  Men ytelses-messig tror jeg ikke det har noe å si mtp. antall iterasjoner .. 

<img width="1076" alt="Skjermbilde 2024-03-06 kl  13 13 46" src="https://github.com/NDLANO/editorial-frontend/assets/26788204/14799efb-bad8-4e27-b149-215d0099a187">